src/easydynamics/sample_model/sample_model.py

```diff
@@ -0,0 +1,277 @@
+
+        if name is None:
+            name = component.name
+        if name in self.list_component_names():
+            raise ValueError(f"Component with name '{name}' already exists.")
+
+        component.name = name
+
+        self.insert(index=len(self), value=component)
+
+    def remove_component(self, name: str):
+        """
+        Remove a model component by name.
+        """
+        # Find index where item.name == name
+        indices = [i for i, item in enumerate(list(self)) if item.name == name]
+        if not indices:
+            raise KeyError(f"No component named '{name}' exists in the model.")
+        del self[indices[0]]
+
+    def list_component_names(self) -> List[str]:
+        """
+        List the names of all components in the model.
+
+        Returns
+        -------
+        List[str]
+            Component names.
+        """
+
+        return [item.name for item in list(self)]
+
+    def clear_components(self):
+        """
+        Remove all components from the model.
+        """
+
+        for _ in range(len(self)):
+            del self[0]
+
+    def normalize_area(self) -> None:
+        # Useful for convolutions.
+        """
+        Normalize the areas of all components so they sum to 1.
+        """
+        if not self.components:
+            raise ValueError("No components in the model to normalize.")
+
+        area_params = []
+        total_area = 0.0
+
+        for component in list(self):
+            if hasattr(component, "area"):
+                area_params.append(component.area)
+                total_area += component.area.value
+            else:
+                warnings.warn(
+                    f"Component '{component.name}' does not have an 'area' attribute and will be skipped in normalization."
+                )
+
+        if total_area == 0:
+            raise ValueError("Total area is zero; cannot normalize.")
+
+        if not np.isfinite(total_area):
+            raise ValueError("Total area is not finite; cannot normalize.")
+
+        for param in area_params:
+            param.value /= total_area
+
+    @property
+    def components(self) -> List[ModelComponent]:
+        """
+        Get the list of components in the SampleModel.
+
+        Returns
+        -------
+        List[ModelComponent]
+        """
+        return list(self)
+
+    @property
+    def unit(self) -> Optional[Union[str, sc.Unit]]:
+        """
+        Get the unit of the SampleModel.
+
+        Returns
+        -------
+        str or sc.Unit or None
+        """
+        return self._unit
+
+    @unit.setter
+    def unit(self, unit_str: str) -> None:
+        raise AttributeError(
+            (
+                f"Unit is read-only. Use convert_unit to change the unit between allowed types "
+                f"or create a new {self.__class__.__name__} with the desired unit."
+            )
+        )  # noqa: E501
+
+    def convert_unit(self, unit: Union[str, sc.Unit]) -> None:
+        """
```

Comment:
Please consider adding the rollback mechanism, which you did in `ModelComponent`.